cpg_workflows/jobs/align.py

```diff
@@ -169,7 +172,21 @@ def align(
     sharded_align_jobs = []
     sorted_bams = []
 
-    if not sharded:  # Just running one alignment job
+    if can_reuse(sorted_bam_path):
+        logging.info(f'{sequencing_group.id} :: Re-using sorted BAM: {sorted_bam_path}')
+        # Its necessary to create this merge_or_align_j object to pass it to finalise_alignment,
+        # and to declare the sorted_bam_path as a resource, so it can be written to the checkpoint.
+        job_attrs = (job_attrs or {}) | dict(label='Reusing sorted bam', tool='Reusing sorted bam')
+        merge_or_align_j = b.new_job('Reusing sorted bam', job_attrs or {})
+        merge_or_align_j.image(config_retrieve(['workflow', 'driver_image']))
+        merge_or_align_j.sorted_bam = b.read_input(str(sorted_bam_path))
```

Comment:
I don't know if this is valid syntax 🤔 
Even if it is I hate the idea of making a fake job just to read a file into the batch. We can definitely re-work the interface here